cmd/rebasehelpers/commitchecker/validate.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/openshift/origin/cmd/rebasehelpers/util"
+)
+
+var UpstreamSummaryPattern = regexp.MustCompile(`UPSTREAM: ([a-z0-9-_]+: )?([0-9]{4,}:|<carry>:|<drop>:)`)
+var UpstreamRevertSummaryPattern = regexp.MustCompile(`UPSTREAM: revert: [a-f0-9]{7,}: ([a-z0-9-_]+: )?([0-9]{4,}:|<carry>:|<drop>:)`)
+
+// ValidateUpstreamCommitsWithoutGodepsChanges returns an error if any
+// upstream commits have no Godeps changes.
+func ValidateUpstreamCommitsWithoutGodepsChanges(commits []util.Commit) error {
+	problemCommits := []util.Commit{}
+	for _, commit := range commits {
+		if commit.HasGodepsChanges() && !commit.DeclaresUpstreamChange() {
+			problemCommits = append(problemCommits, commit)
+		}
+	}
+	if len(problemCommits) > 0 {
+		label := "No commits were marked UPSTREAM, but the following Godeps changes were detected"
```

Comment:
This isn't entirely accurate.  I could have an UPSTREAM commit and a normal commit and make godep changes in the normal commit.  This message would then appear.